### ISSUE TITLE
Getting a very vague "VCF Parser: could not open file"

### ISSUE THREAD

**Issue by bhospelhorn** [fritzsedlazeck/SURVIVOR#114]
Hey!

I keep getting this vague message on merge -- "VCF Parser: could not open file: (file path)"... for files that definitely exist and definitely are properly formatted VCFs. I cannot figure out why, but it will not take any of my files. Tried this in both cygwin (using both cygwin style and windows style paths) and Fedora - didn't work. I'd been using SURVIVOR for a couple years on the cluster at work and never had this issue...

Is this a common issue for people trying to compile/run this from home? Or is there some quirk to the parser that requires I modify the files in some way?

**Comment by fritzsedlazeck** [comment]
Hi,
thanks for reaching out. 
1. Is there an empty line maybe in the files file? 
2. Can you confirm that an ls on the file path as it is in the files file works?
3. Do you have read permissions on these files?
4. Can you past the full output?

Thanks
Fritz

**Comment by bhospelhorn** [comment]
Sure!

I didn't see any empty lines in the files. Also tried adding a newline at the end because some parsers insist upon that, and that didn't work either.

Here's the output + an ls:
`[bhospelhorn@localhost svproject]$ DataDir="/home/bhospelhorn/Documents/data/svproject/GIAB"
[bhospelhorn@localhost svproject]$ SurvivorExe="/home/bhospelhorn/Documents/Git/SURVIVOR/Debug/SURVIVOR"
[bhospelhorn@localhost svproject]$ 
[bhospelhorn@localhost svproject]$ MergeInList=${DataDir}"/NA12878/Analysis/merge2020/paths.txt"
[bhospelhorn@localhost svproject]$ MergeOutStem=${DataDir}/"NA12878/Analysis/merge2020/NA12878_"
[bhospelhorn@localhost svproject]$ 
[bhospelhorn@localhost svproject]$ MaxDistBool=0
[bhospelhorn@localhost svproject]$ ConsiderType=0
[bhospelhorn@localhost svproject]$ ConsiderStrand=1
[bhospelhorn@localhost svproject]$ DistEstOnSize=1
[bhospelhorn@localhost svproject]$ ${SurvivorExe} merge ${MergeInList} ${MaxDistBool} 2 ${ConsiderType} ${ConsiderStrand} ${DistEstOnSize} 50 ${MergeOutStem}"Intersect_min2.vcf"
VCF Parser: could not open file: /home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_mtsinai.vcf
[bhospelhorn@localhost svproject]$ ls -l /home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_mtsinai.vcf
-rw-rw----. 1 bhospelhorn bhospelhorn 4988638 Mar 20 12:30 /home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_mtsinai.vcf`

And paths.txt consists of
`/home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_mtsinai.vcf
/home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_mills_liftover.vcf
/home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_svclassify_del.vcf
/home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_svclassify_ins.vcf
/home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_10x.vcf
/home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/NA12878_dbvar.vcf`

Just playing with merging some of the GIAB callsets together. Some of the VCFs I generated, some I used as-is... none of them appear to work - it just rejects the first file in paths.txt and exits.

**Comment by fritzsedlazeck** [comment]
Can you confirm that there is no space or so after the name in the file? 

I usually use an 
`ls /home/bhospelhorn/Documents/data/svproject/GIAB/NA12878/Analysis/merge2020/*vcf > files
`
This may sound stupid but many people and I are using this almost daily so something seems to be going on since otherwise everyone would complain.. 
Thanks
Fritz

**Comment by bhospelhorn** [comment]
Yup, there it is.
Checked a hex editor - paths.txt still had bloody carriage returns. Thought repasting it line by line in the Linux text editor would take care of that but _apparently not_ haha... Wouldn't THINK I would need to constantly be checking my text files in a hex editor.... gr...

Ugh, thank you so much for your help still. Since the message said there was a parser error (sometimes "can't open" is also used for "can't read"), I just really didn't know if it was a path formatting issue or if there really was something odd about these files or running it on Fedora or something.

I'm so sorry for bugging you with this - I got really stumped since I couldn't figure out what was wrong.
